orangecontrib/text/widgets/owpreprocess.py

```diff
@@ -1,70 +1,706 @@
-from PyQt4 import QtCore, QtGui
+import os
 
+from PyQt4.QtCore import (pyqtSignal as Signal, pyqtSlot as Slot)
+from PyQt4.QtGui import (QWidget, QLabel, QHBoxLayout, QVBoxLayout,
+                         QButtonGroup, QRadioButton, QSizePolicy, QFrame,
+                         QComboBox, QPushButton, QStyle, QApplication,
+                         QFileDialog, QLineEdit, QCheckBox)
+from nltk.corpus import stopwords
+
+from Orange.widgets import gui, settings
 from Orange.widgets.widget import OWWidget
-from Orange.widgets.settings import Setting
-from Orange.widgets import gui
+from orangecontrib.text.corpus import Corpus
+from orangecontrib.text.preprocess import (PorterStemmer as PS,
+                                           Lemmatizer as LM,
+                                           SnowballStemmer as SS)
 from orangecontrib.text.preprocess import Preprocessor
 
 
+def _i(name, icon_path='icons'):
+    widget_path = os.path.dirname(os.path.abspath(__file__))
+    return os.path.join(widget_path, icon_path, name)
+
+
+class Input:
+    CORPUS = 'Corpus'
+
+
 class Output:
-    PREPROCESSOR = "Preprocessor"
+    PP_CORPUS = 'Corpus'
+
+
+class PreprocessorModule(QWidget):
+    """The base widget for the pre-processing modules."""
+
+    change_signal = Signal()  # Emitted when the settings are changed.
+    # Emitted when the module has a message to display in the main widget.
+    error_signal = Signal(str)
+    enabled = False  # If the module is enabled.
+
+    def __init__(self, title, toggle_enabled, is_enabled):
+        super(PreprocessorModule, self).__init__()
```

Comment:
Could you simplify this to `super().__init__()`? Is there a need for passing of the `PreprocessorModule`?